### PR TITLE
OZ-707: E2E tests verifying Odoo groups are synced to Keycloak.

### PR DESCRIPTION
**Pull request by kdaud**
Ticket: https://mekomsolutions.atlassian.net/browse/OZ-707

This PR adds E2E tests for the flows between Keycloak and Odoo.
- Existing Odoo groups create the corresponding Keycloak role.
- Creating new group in Odoo creates the role in Keycloak.
- Renaming a group in Odoo updates the corresponding role in Keycloak.

### CODE REVIEW

File: e2e/tests/keycloak-superset-flows.spec.ts [hunk 1]
```diff
@@ -14,6 +14,34 @@ test.beforeEach(async ({ page }) => {
   superset = new Superset(page);
 });
 
+test('Logging out from Superset logs out the user from Keycloak.', async ({ page }) => {
```
Comment (corneliouzbett) [inline e2e/tests/keycloak-superset-flows.spec.ts:17]: Probably, you should mention something like "_Logging out from Superset ends the session in Keycloak and logs out the user_"

File: e2e/utils/functions/odoo.ts [hunk 2]
```diff
@@ -55,6 +64,66 @@ export class Odoo {
...
+  }
+
+  async createGroup() {
+    await this.page.getByRole('button', { name: /create/i }).click();
+    await this.page.getByLabel(/application/i).click();
+    await expect(this.page.getByText(/accounting/i)).toBeVisible();
+    await this.page.getByText(/accounting/i).click();
+    await this.page.getByLabel(/name/i).fill(`${randomOdooGroupName.groupName}`);
+    await this.page.getByRole('button', { name: /save/i }).click();
+    await delay(240000);
```
Comment (corneliouzbett) [inline e2e/utils/functions/odoo.ts:94]: That's a big delay?

File: e2e/utils/functions/openmrs.ts [hunk 3]
```diff
@@ -384,7 +384,7 @@ export class OpenMRS {
     await this.page.getByLabel('Application: Writes Clinical Notes').check();
     await this.page.getByRole('button', { name: 'Save Role' }).click();
     await expect(this.page.getByText('Role saved')).toBeVisible();
-    await delay(50000);
+    await delay(160000);
```
Comment (corneliouzbett) [inline e2e/utils/functions/openmrs.ts:387]: That's a huge bump, what does it mean?

**Review by corneliouzbett — APPROVED** [review]
Few general comments. Otherwise, this LGTM!

File: e2e/utils/functions/odoo.ts (hunk 2 above)
Comment (kdaud) [reply e2e/utils/functions/odoo.ts:94]: Manually, it takes 2 to 3 minutes for the sync to occur between Odoo and Keycloak. In the testing environment, it takes an additional minute for the test to pass.

Comment (corneliouzbett) [reply e2e/utils/functions/odoo.ts:94]: Yes, I remember the delay. Make sense!

File: e2e/utils/functions/openmrs.ts (hunk 3 above)
Comment (kdaud) [reply e2e/utils/functions/openmrs.ts:387]: It takes some 1 to 2 minutes the eip client for openmrs-keycloak to sync.

**Pull request merged**